homeassistant/components/utility_meter/sensor.py

```diff
@@ -147,7 +149,44 @@ async def async_setup_platform(
     )
 
 
-class UtilityMeterSensor(RestoreEntity, SensorEntity):
+@dataclass
+class UtilitySensorExtraStoredData(SensorExtraStoredData):
+    """Object to hold extra stored data."""
+
+    native_value: Decimal  # type: ignore[assignment]
+    last_period: Decimal
+    last_reset: datetime
+    status: str
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return a dict representation of the utility sensor data."""
+        data = super().as_dict()
+        data["native_value"] = str(self.native_value)
+        data["last_period"] = str(self.last_period)
+        data["last_reset"] = self.last_reset.isoformat()
+        data["status"] = self.status
+
+        return data
+
+    @classmethod
+    def from_dict(cls, restored: dict[str, Any]) -> UtilitySensorExtraStoredData | None:
+        """Initialize a stored sensor state from a dict."""
+        try:
+            native_value = Decimal(restored["native_value"])
+        except InvalidOperation:
+            native_value = None
+        native_unit_of_measurement = restored["native_unit_of_measurement"]
+        try:
+            last_period = Decimal(restored["last_period"])
+        except InvalidOperation:
+            last_period = Decimal(0)
+        last_reset = dt_util.parse_datetime(restored["last_reset"])
+        status = restored["status"]
```

Comment:
If the data is bad, let's just return None, instead of partially restoring to some broken state:
```suggestion
        try:
            native_value = Decimal(restored["native_value"])
            native_unit_of_measurement = restored["native_unit_of_measurement"]
            last_period = Decimal(restored["last_period"])
            last_reset = dt_util.parse_datetime(restored["last_reset"])
            status = restored["status"]
        except (InvalidOperation, KeyError, TypeError):
            return None
        if last_reset is None:
            return None
```

Reply:
last_period and last_reset will be None up to the 1st reset, yet all other aspects will be fine